pkg/storage/fs/ocis/upload.go

```diff
@@ -0,0 +1,499 @@
+// Copyright 2018-2020 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package ocis
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/logger"
+	"github.com/cs3org/reva/pkg/user"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/pkg/xattr"
+	"github.com/rs/zerolog/log"
+	tusd "github.com/tus/tusd/pkg/handler"
+)
+
+var defaultFilePerm = os.FileMode(0664)
+
+// TODO deprecated ... use tus
+
+func (fs *ocisfs) Upload(ctx context.Context, ref *provider.Reference, r io.ReadCloser) error {
+
+	node, err := fs.pw.NodeFromResource(ctx, ref)
+	if err != nil {
+		return err
+	}
+
+	if node.ID == "" {
+		node.ID = uuid.New().String()
+	}
+
+	nodePath := filepath.Join(fs.conf.Root, "nodes", node.ID)
+
+	tmp, err := ioutil.TempFile(nodePath, "._reva_atomic_upload")
+	if err != nil {
+		return errors.Wrap(err, "ocisfs: error creating tmp fn at "+nodePath)
+	}
+
+	_, err = io.Copy(tmp, r)
+	if err != nil {
+		return errors.Wrap(err, "ocisfs: error writing to tmp file "+tmp.Name())
```

Comment:
delete temp file on error or does this happen automatically ? how often ?

Reply:
hm ... the put is deprecated ... but yeah we should delete the file in this case